lib/rules/arrow-parens.js

```diff
@@ -80,29 +80,46 @@ module.exports = {
          * @returns {void}
          */
         function parens(node) {
-            const isAsync = node.async;
-            const firstTokenOfParam = sourceCode.getFirstToken(node, isAsync ? 1 : 0);
+            const firstToken = sourceCode.getFirstToken(node);
+            const arrowToken = sourceCode.getTokens(node).find(astUtils.isArrowToken);
```

Comment:
This code might find an unexpected arrow token before the one that it's supposed to (e.g. in a default value for a param)? 

While the final outcome will be probably the same, it might be more clear to get the first arrow token before the body, as it's implemented in `astUtils.getFunctionHeadLoc`?